src/foam/u2/view/ReadReferenceView.js

```diff
@@ -9,36 +9,138 @@ foam.CLASS({
+                               /                       \                       /                   \
+                    Is the menu dao Menu and      disable link           enable link           disable link
+                    dao Menu daoKey matches                           link to daoSummary
+                    prop.targetDAOKey ?
+                        /      \  n
+                     y /        - - - - - -
+                      /                     \
+                enable link            enable link
+              link to daoSummary     link to the menu
+
+    -- Where to set enableLink and menuKeys --
+        - enableLink and menuKeys can be set directly to reference property or
+          they can be passed to the view. 
+          Warning: Do not try to do both or it may result in unexpected behaviours.
+    
+        1. setting properties on reference property
+
+        {
+          class: 'Reference',
+          of: 'foam.nanos.auth.Group',
+          name: 'group',
+          menuKeys: [
+            'someMenuId',
+            'someMenuId2'
+          ]
+        }
+
+        2. passing properties to view
+
+        class: 'Reference',
+        of: 'foam.nanos.auth.Group',
+        name: 'group',
+        view: {
+          class: 'foam.u2.view.ReadReferenceView',
+          menuKeys: [
+            'someMenuId',
+            'someMenuId2'
+          ]
+        }
+  `,
 
   requires: [
     'foam.comics.v2.DAOControllerConfig',
     'foam.u2.detail.SectionedDetailView',
     'foam.u2.view.ReferenceCitationView'
   ],
 
-  axioms: [
-    foam.pattern.Faceted.create()
-  ],
+  axioms: [foam.pattern.Faceted.create()],
 
   properties: [
     'obj',
     {
       name: 'of',
-      expression: function(obj) { return obj.cls_; }
+      expression: function (obj) {
+        return obj.cls_;
+      }
     },
     'prop',
+    {
+      class: 'String',
+      name: 'linkTo',
+      documentation: `
+        A reference view anchor link.
+        This will be set to either 'daoSummary', a menu in menuKeys, or an empty string.
+        If set to 'daoSummary', the link will be to reference property's dao summary view.
+        If set to a menu, the link will be to this menu.
+        If set to an empty string, the link will be disabled.
+      `
+    },
     {
       class: 'Boolean',
       name: 'enableLink',
-      documentation: 'Create the reference view as an anchor link to the reference\'s DetailView.',
+      documentation: `
+        Create the reference view as an anchor link to the reference's DetailView or provided menu.
+      `,
       value: true
+    },
+    {
+      name: 'menuKeys',
+      documentation: `
+        A list of menu ids.
+        The link will reference to the first menu to which you have permission
+        in this list. If no menus are permissioned, the link will be disabled.
+      `,
+      expression: function(prop) {
```

Comment:
I don't like this. We're hard-coding nanopay application specific functionality into generic FOAM code. This will be confusing for developers as some targetDAOKeys will magically work while others won't.